File: DotNet/Opertat-Core/NeuralNetworkFlash.cs
using System;
using System.Collections.Generic;
using MathNet.Numerics.LinearAlgebra;

namespace Photon.NeuralNetwork.Opertat
{
    public class NeuralNetworkFlash
    {
        internal readonly Vector<double>[] SignalsSum;
        internal readonly Vector<double>[] InputSignals;
        private readonly List<Vector<double>[]> SignalsExtra;
        public double[] ResultSignals { get; internal set; }
        public NeuralNetworkFlash(int size)
        {
            SignalsSum = new Vector<double>[size];
            InputSignals = new Vector<double>[size + 1];
            SignalsExtra = new List<Vector<double>[]>();
        }

        public Vector<double>[] this[int index]
        {
            get
            {
                while (SignalsExtra.Count <= index)
                    SignalsExtra.Add(new Vector<double>[SignalsSum.Length]);
                return SignalsExtra[index];
            }
            set
            {
                while (SignalsExtra.Count <= index)
                    SignalsExtra.Add(new Vector<double>[SignalsSum.Length]);
                SignalsExtra[index] = value;
            }
        }
    }
}

File: Java/Opertat-Core/src/Photon/NeuralNetwork/Opertat/NeuralNetworkFlash.java
package Photon.NeuralNetwork.Opertat;

import java.util.ArrayList;
import java.util.List;

import org.ejml.data.DMatrixRMaj;

public class NeuralNetworkFlash {

	final DMatrixRMaj[] signals_sum;
	final DMatrixRMaj[] input_signals;
	private final List<DMatrixRMaj[]> signals_extra;
	double[] result_signals;

    public NeuralNetworkFlash(int size)
    {
    	signals_sum = new DMatrixRMaj[size];
    	input_signals = new DMatrixRMaj[size + 1];
    	signals_extra = new ArrayList<DMatrixRMaj[]>();
    }

	public double[] ResultSignals() {
		return result_signals;
	}

}
